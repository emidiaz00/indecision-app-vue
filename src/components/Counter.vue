<template>
    <div>
        <h1>{{ customTitle }}</h1>
        
        <p>  {{counter}} <sup>2</sup>  = {{ getSquareCounter }} </p>
        <button v-on:click="addCounter">+</button>
        <button v-on:click="subtractCounter">-</button>
        <button class="btn btn-danger" v-on:click="resetCounter">RESET</button>
        
    </div>
</template>


<script>
export default {
    props: {
        title: String,
        start: {
            type: Number,
            default: 100,
            // required true
            validator (value) {
            return value > 103
        }
        },
        
    },
    data() {
        return {
            counter: this.start
        }
    },
    methods: {
        addCounter() {
            this.counter++
        },
        subtractCounter() {
            this.counter--
        },
        resetCounter() {
            this.counter = 1
        }
    },
    computed: {
        getSquareCounter() {
            return this.counter + this.counter
        },
        customTitle() {
            if (!this.title) {
                return this.counter
            }
        }
    }
}
</script>

<style>
p {
    color: black;
    font-weight: bold;
    font-size: 40px;
    margin: 0 auto;
}
button {
    background-color: #497174;
    border: none;
    border-radius: 2px;
    padding: 4px 38px;
    margin: 10px;
    color: #EFF5F5;
    font-weight: bold;
    font-size: 18px;
    cursor: pointer;
    margin: 6px;
}
</style>