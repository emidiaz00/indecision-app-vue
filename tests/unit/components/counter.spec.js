import Counter from '@/components/Counter'
import { shallowMount, mount } from '@vue/test-utils'

describe('Counter component', () => {
    test('should be equal to snapshot', () => { 
        const wrapper = shallowMount(Counter)
        expect(wrapper.html()).toMatchSnapshot()
     })
     test('should increase value of counter', () => { 
        const wrapper = shallowMount(Counter)
        const increaseBtn = wrapper.find('button')

        increaseBtn.trigger('click')
        const value = wrapper.find('[data-testid="counter"]').text()
        expect(value).toBe('100')
      })
});

