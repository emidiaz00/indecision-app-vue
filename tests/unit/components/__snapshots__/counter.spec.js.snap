// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter component should be equal to snapshot 1`] = `
<div>
  <h1>100</h1>
  <p>100<sup>2</sup> = 200</p><button>+</button><button>-</button><button class="btn btn-danger">RESET</button>
</div>
`;
